src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,51 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        if (country == null) {
+            throw new RuntimeException("unacceptable data");
+        }
+
+        if (country.getId() != null && get(country.getId()).isPresent()) {
+            throw new RuntimeException(country + "already existed in database");
+        }
+
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingExeption("adding " + country + " into database failed");
```

Comment:
The exception thrown here uses `DataProcessingExeption`, which contains a typo. It should be `DataProcessingException` to ensure the correct exception class is used.